### PR TITLE
FIX for: Logs being recorded even with debug option disabled

### PR DESCRIPTION
**Pull request by wilmerlm11**
Also reported here: https://wordpress.org/support/topic/logs-being-recorded-even-with-debug-option-disabled/

There are are messages spamming the logs, even with the debug option being disabled in the plugin config.

It comes from:
facebook-for-woocommerce\includes\Handlers\Connection.php

Line 394: private function retrieve_page_access_token( $page_id )
One of them is:
facebook_for_woocommerce()->log( ‘Retrieving page access token’ );

What I saw is that the log function has not been overridden to check the debug option as it’s done in the similar function log_api_request.

I'll create a PR with the suggested change to fix that.

Hopefully it can make the next release.
It’s annoying the need to keep cleaning up my logs if I chose to not log anything from this plugin.

### CODE REVIEW

File: class-wc-facebookcommerce.php
Comment (jconroy) [inline class-wc-facebookcommerce.php:374]: Can you update to use `2.3.1-dev.1`

**Review by jconroy — CHANGES_REQUESTED** [review]
Thanks so much for taking the time to submit a PR @wilmerlm11 , just one really quick comment before a proper test and review

And just to confirm - the way to replicate your issue is to simply disable debugging (in the plugin) and we should still see a facebook specific log file  generated and populated.

After this patch that won't happen

Is that your expectation?

**Review by wilmerlm11 — COMMENTED** [review]
Updated "since" version as requested :)

**Comment by wilmerlm11** [comment]
Hi @jconroy ,
thank you for reviewing it!
Your test case is perfect. That's exactly it.
After the fix, you should only see those logs recorded through facebook_for_woocommerce()->log( ), IF the debugging option is enabled in the plugin admin config.
Regards,
Wil

**Comment by jconroy** [comment]
Just making a note that I've replicated the issue

Screen Shot 2021-02-24 at 8 42 52 pm

**Comment by wilmerlm11** [comment]
@jconroy , I've just noticed 2.3.1 is out (without this fix :( ).
Should we keep 2.3.1 in the "since" version?
Any expectation to have this fix merged, since it's a simple one?

Thanks!!

**Comment by jconroy** [comment]
Sorry we didn't get a chance to test properly before last week's release. 

> Any expectation to have this fix merged, since it's a simple one?

Absoolutely, appreciate the contribution - we just need to find some time to double check things.

File: class-wc-facebookcommerce.php
Comment (rcstr) [inline class-wc-facebookcommerce.php:374]: We'll need to update the @`since` tag to `2.3.3` before merging.

**Review by rcstr — CHANGES_REQUESTED** [review]
@wilmerlm11 - I have tested your changes, and it works as expected. Thank you for submitting this PR.

I have requested changes because we need to update the `version` before we merge

File: class-wc-facebookcommerce.php
Comment (wilmerlm11) [reply class-wc-facebookcommerce.php]: Hi @rcstr !
Tks for validating the fix.
I've just updated the since first as requested.

**Review by rcstr — APPROVED** [review]
thank you @wilmerlm11!

**Comment by wilmerlm11** [comment]
Hi @jconroy !
Does it need you approval yet?
I'm afraid the release train will pass and I'll have to update the "since" version once again :/
Regards

**Comment by jconroy** [comment]
All merged into the 2.3.3 release branch @wilmerlm11 - thank you!

**Pull request merged**
